### ISSUE TITLE
Proxy support for AWS SDK

### ISSUE THREAD

**Issue by aureliomarcoag** [promptfoo/promptfoo#1010]
Proxy support was added in https://github.com/promptfoo/promptfoo/pull/299, but not for the AWS SDK, which requires additional configuration.

For the AWS SDK, a NodeHttpHandler (from `@smithy/node-http-handler`) needs to be passed to BedrockRuntime, e.g.:
```
import { ProxyAgent } from 'proxy-agent';
import { NodeHttpHandler } from '@smithy/node-http-handler';
import { BedrockRuntime } from '@aws-sdk/client-bedrock-runtime';
...
const handler = new NodeHttpHandler({httpsAgent: new ProxyAgent()});
const bedrock = new BedrockRuntime({ region: this.getRegion(), requestHandler: handler });
```

Note: NodeHttpHandler supports additional configuration values, but I don't know whether it's necessary to allow customization of those (in my tests simply using the `httpsAgent` was enough).

Affected file: [bedrock.ts:453](https://github.com/promptfoo/promptfoo/blob/main/src/providers/bedrock.ts#L453)

**Comment by aureliomarcoag** [comment]
Cheers @mldangelo thanks for taking care of this one so quickly 😄 
https://github.com/promptfoo/promptfoo/pull/1021

I downloaded the code for the commit https://github.com/promptfoo/promptfoo/tree/11ccd73ddaa63ea7142f6ae7269284c3030d9d4d built it and ran it in our pipeline behind a proxy, the issue is fixed for us.

Side note: I checked the PR just now, I think it's not necessary to check for the proxy-related env vars, since ProxyAgent takes care of it internally. In any case, it's working for us now, thanks!
